Source/JavaScriptCore/heap/MarkedBlockInlines.h

```diff
@@ -282,8 +282,9 @@ void MarkedBlock::Handle::specializedSweep(FreeList* freeList, MarkedBlock::Hand
         
         char* startOfLastCell = static_cast<char*>(cellAlign(block.atoms() + m_endAtom - 1));
         char* payloadEnd = startOfLastCell + cellSize;
-        RELEASE_ASSERT(payloadEnd - MarkedBlock::blockSize <= bitwise_cast<char*>(&block));
         char* payloadBegin = bitwise_cast<char*>(block.atoms());
+        if (UNLIKELY(!(payloadEnd - MarkedBlock::blockSize <= bitwise_cast<char*>(&block))))
+            CRASH_WITH_INFO(payloadBegin, payloadEnd, &block, cellSize, m_endAtom);
```

Comment:
Instead of using an if statement, you can simply express this as:
`RELEASE_ASSERT(payloadEnd - MarkedBlock::blockSize <= bitwise_cast<char*>(&block), payloadBegin, payloadEnd, &block, cellSize, m_endAtom); `

Reply:
That's nice!! Changed.